### PR TITLE
picotorrent: Add file type association

### PR DESCRIPTION
**Pull request by BlackPowerade**
Adds filetype association for .torrent and magnet links. The `post_install` is created from vscode's.
Will change `https://raw.githubusercontent.com/BlackPowerade/scoop-extras/picotorrent-regs/` to `https://raw.githubusercontent.com/lukesampson/scoop-extras/master/` if approved.

### CODE REVIEW

File: bucket/picotorrent.json [hunk 1]
```diff
@@ -25,12 +41,32 @@
...
     "persist": [
         "Torrents",
         "Session.dat",
-        "PicoTorrent.json"
+        "PicoTorrent.json",
+        "PicoTorrent.sqlite"
+    ],
+    "post_install": [
+        "if (Test-Path \"$dir\\picotorrent-install-associations.reg\") {",
+        "  $picopath = \"$dir\\PicoTorrent.exe\".Replace('\\', '\\\\')",
```
Comment (Ash258) [inline bucket/picotorrent.json]: indent with 4 spaces ...

```suggestion
        "    $picopath = \"$dir\\PicoTorrent.exe\".Replace('\\', '\\\\')",
```

File: bucket/picotorrent.json [hunk 2]
```diff
@@ -42,7 +42,7 @@
     "pre_install": [
         "if (!(Test-Path \"$persist_dir\\Session.dat\")) { Set-Content \"$dir\\Session.dat\" $null -Encoding Ascii }",
         "if (!(Test-Path \"$persist_dir\\PicoTorrent.json\")) { Set-Content \"$dir\\PicoTorrent.json\" '{}' -Encoding Ascii }",
-        "if (!(Test-Path \"$persist_dir\\PicoTorrent.sqlite\")) { New-Item \"$dir\\PicoTorrent.sqlite\" -ItemType File | Out-Null }"
+        "if (!(Test-Path \"$persist_dir\\PicoTorrent.sqlite\")) { New-Item \"$dir\\PicoTorrent.sqlite\" -Encoding Ascii | Out-Null }"
```
Comment (Ash258) [inline bucket/picotorrent.json]: New-item do not have encoding ...

**Comment by BlackPowerade** [comment]
Forgot to mention, but another change here is `Picotorrent.sqlite` being added to the persist. This is what keeps the data for torrent progress, active torrents, where they are located/being saved to, etc.

File: bucket/picotorrent.json [hunk 3]
```diff
@@ -8,12 +8,28 @@
     "version": "0.18.0",
     "architecture": {
         "64bit": {
-            "url": "https://github.com/picotorrent/picotorrent/releases/download/v0.18.0/PicoTorrent-0.18.0-x64.zip",
-            "hash": "80da9bd74845ad8f13b9ce604ccb91047fee90c2dadce0607d48fb2c2c106491"
+            "url": [
+                "https://github.com/picotorrent/picotorrent/releases/download/v0.18.0/PicoTorrent-0.18.0-x64.zip",
+                "https://raw.githubusercontent.com/lukesampson/scoop-extras/picotorrent-regs/scripts/picotorrent/install-associations.reg",
```
Comment (linsui) [inline bucket/picotorrent.json]:
```suggestion
                "https://raw.githubusercontent.com/lukesampson/scoop-extras/master/scripts/picotorrent/install-associations.reg",
```

**Comment by Ash258** [comment]
Anyone tested actual installation, persisting of precreated sqlite file and registry files are correct and working?

I do not have nearby device, where I can manipulate with registry in this manner.

**Comment by BlackPowerade** [comment]
Testing in VM works alright

**Comment by BlackPowerade** [comment]
Everything seems to work fine here. The only thing I cannot verify is if the hashes for the registry files will still match after being merged.

**Pull request merged**